default/python/AI/Behavior/Character.py

```diff
@@ -0,0 +1,517 @@
+                                           "LRN_TRANSLING_THT": fo.aggression.typical,
+                                           "LRN_PSIONICS": fo.aggression.typical,
+                                           "LRN_DISTRIB_THOUGHT": fo.aggression.typical,
+                                           "LRN_QUANT_NET": fo.aggression.typical,
+                                           "PRO_SINGULAR_GEN": fo.aggression.typical,
+                                           "LRN_TRANSCEND": fo.aggression.typical}
+
+    def may_research_tech_classic(self, tech):
+        return (type(self).tech_lower_threshold_classic_static.get(tech, fo.aggression.beginner)
+                <= self.aggression <=
+                type(self).tech_upper_threshold_classic_static.get(tech, fo.aggression.maniacal))
+
+
+    def attitude_to_empire(self, other_empire_id, diplomatic_logs):
+        # TODO: In other behaviors consider proximity, competitive
+        # needs, relations with other empires, past history with this
+        # empire, etc.
+        # in the meantime, somewhat random
+        # TODO: Move the diplomatic log portion of this behavior back
+        # into diplomacy where it belongs.
+        if self.aggression == fo.aggression.maniacal:
+            return -9
+        if self.aggression == fo.aggression.beginner:
+            return 9
+        log_index = (other_empire_id, fo.empireID())
+        num_peace_requests = len(diplomatic_logs.get('peace_requests', {}).get(log_index, []))
+        num_war_declarations = len(diplomatic_logs.get('war_declarations', {}).get(log_index, []))
+        # Too many requests for peace irritate the AI, as do any war declarations
+        irritation = (self.aggression * (2.0 + num_peace_requests / 10.0 + 2.0 * num_war_declarations) + 0.5)
+        attitude = 10 * random.random() - irritation
+        return min(10, max(-10, attitude))
+
+
+class EmpireIDBehavior(Behavior):
+    """A behavior that models empire id influence.
+    Mostly some modulo 2 effects."""
+    # TODO: Stop using modulo empire id as a character component.
+    # Empire id is a game mechanic.  It is not something that a player
+    # can describe, "Look the 'Continuum' is behaving like a 1 modulo 2 character."
+
+    def __init__(self, empire_id, aggression):
+        print "EmpireIDBehavior initialized."
+        self.id = empire_id
+        self.aggression = aggression  # TODO remove when old research style get_research_index is removed
+
+    @property
+    def key(self):
+        return self.id % 2
+
+    def preferred_research_cutoff(self, alternatives):
+        if not alternatives:
+            return None
+        return alternatives[self.id % 2] if len(alternatives) >= 2 else alternatives[0]
+
+    def preferred_colonization_portion(self, alternatives):
+        if not alternatives:
+            return None
+        return alternatives[self.id % 2] if len(alternatives) >= 2 else alternatives[0]
```

Comment:
If the original intent is that there are n alternatives to choose from, then simply use
```
try:
    return alternatives[self.id % n]
catch IndexError as e:
    print_error(e)
    return None
```
which may be implemented as private class method.

Requires the additional import
```from freeorion_tools import print_error```